packages/engine/Source/Scene/Model/RuntimeModelInstancingPipelineStage.js

```diff
@@ -0,0 +1,256 @@
+import Cartesian3 from "../../Core/Cartesian3.js";
+import combine from "../../Core/combine.js";
+import ComponentDatatype from "../../Core/ComponentDatatype.js";
+import defined from "../../Core/defined.js";
+import EncodedCartesian3 from "../../Core/EncodedCartesian3.js";
+import Matrix4 from "../../Core/Matrix4.js";
+import Buffer from "../../Renderer/Buffer.js";
+import BufferUsage from "../../Renderer/BufferUsage.js";
+import ShaderDestination from "../../Renderer/ShaderDestination.js";
+import InstancingStageCommon from "../../Shaders/Model/InstancingStageCommon.js";
+import RuntimeModelInstancingPipelineStageVS from "../../Shaders/Model/RuntimeModelInstancingPipelineStageVS.js";
+
+const nodeTransformScratch = new Matrix4();
+
+/**
+ * The instancing pipeline stage is responsible for handling GPU mesh instancing at the node
+ * level for model instances specified using <code>ModelInstance</code>
+ *
+ * @namespace RuntimeModelInstancingPipelineStage
+ * @private
+ */
+const RuntimeModelInstancingPipelineStage = {
+  name: "RuntimeModelInstancingPipelineStage", // Helps with debugging
+};
+
+/**
+ * Process a node. This modifies the following parts of the render resources:
```

Comment:
This was copied over from the previous instancing stage, and will need to be updated to reflect the actual workflow.